app/Platform/Actions/BuildHubBreadcrumbsAction.php

```diff
@@ -0,0 +1,516 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\GameSet;
+use App\Platform\Data\GameSetData;
+use App\Platform\Enums\GameSetType;
+use Illuminate\Support\Facades\Cache;
+
+/**
+ * Builds breadcrumb navigation paths for hub (game set) hierarchies.
+ *
+ * This action  generates breadcrumb navigation paths for hubs, handling various
+ * special cases and hierarchical relationships. It supports different types of
+ * game sets including themes, difficulty levels, and various hub types.
+ *
+ * - Caches breadcrumb paths with fresh/stale timing.
+ * - Handles special cases for Theme and Difficulty hubs.
+ * - Supports derived hierarchical navigation through hub types.
+ * - Manages hub relationships via predefined mappings.
+ */
+class BuildHubBreadcrumbsAction
+{
+    /**
+     * Cache configuration for breadcrumb data.
+     */
+    private const FRESH_SECONDS = 600; // 10 minutes
+    private const STALE_SECONDS = 1200; // 20 minutes
+
+    /**
+     * Common prefix for central hub entries.
+     */
+    private const PREFIX_CENTRAL = '[Central - ';
+
+    /**
+     * Defines the hierarchical relationships between different hub types.
+     * The key represents the child type, and the value represents the parent type.
+     *
+     * @var array<string, string>
+     */
+    private const HUB_HIERARCHY = [
+        'Custom Awards' => 'Central',
+        'Feature' => 'Central',
+        'Game Mechanic' => 'Central',
+        'Meta' => 'Central',
+        'Meta|QA' => 'Meta',
+        'Meta|DevComp' => 'Meta',
+        'Regional' => 'Central',
+        'Series' => 'Central',
+        'Subseries' => 'Series',
+        'Technical' => 'Central',
```

Comment:
Here's another one-off:
![image](https://github.com/user-attachments/assets/e304df07-47e3-46fb-9e71-d643d6b18fcb)

The DevQuest sets hubs aren't parented to the the DevQuest event